PushFixed function for static dataset size

Ropes opened the pull request:
Instead of Push simply appending data points to ever expanding Go slices; this change
will utilize GoVector's PushFixed function to keep the array size constant as data points 
are added by dropping the oldest event. The goal is to keep memory allocations for long 
running processes constantly evaluating anomalies consistent.

eg

```
anomalyzer.Data: [ 1.0, 2.0, 3.0] 
err := anomalyzer.PushFixed(5.0)
anomalyzer.Data: [2.0, 3.0, 5.0]
```

Mixing `Push` and `FixedPush` for a anomalyzer data set will result in failures. The problem stems from the `len` of an array changing, but the `cap` not also being updated. There might be a easy fix another change will need to be made to GoVector.

Ropes committed:
New PushFixed function to add data points to fixed size arrays

Instead of push simply appending data points to expanding Go slices; this change
will allow a rolling window dataset to be analyzed for anomalous
signals.

Ropes committed:
Plumbing through change to GoVectors PushFixed to handle returned error

Ropes committed:
Minor clarity update to README for the NA const

Ropes committed:
Updating readme and code with warnings about mixing Push and FixedPush

Mixing Push and FixedPush will result in errors, tests prove this
problem(not committed yet). I'll leave that for another day when there's
free time.

Ropes committed:
Fixing spacing typo

allisonmorgan commented:
I really like this method and I almost wonder if it would be useful enough for others that an `AnomalyzerConf` option should be something like `Cap`. So that if `Cap` is unspecified, then we just default to the `Push()` option, otherwise if `len(data) > Cap` we start to `PushFixed()` (or whatever doesn't result in failures). @drewlanenga?

Ropes commented:
I agree, I'd like to have a safe usage of both intermixed, or configurable like you state. However it might just need a re-evaluation of my GoVector code to fix the problem.

Ropes committed:
Tests to prove GoVector's: `pushfixed-handles-exception` branch changes

Changed the PushFixed array code to not break if the array was expanded
by an external call to the standard Push function. The two functions can
now be called in a mixed fashion, and PushFixed will still attempt to
use as little memory as possible.

Ropes commented:
@drewlanenga will need to merge [this](https://github.com/drewlanenga/govector/pull/8) PR into GoVector in order for the tests I added to pass.

Ropes commented:
Took me a little bit to figure out why it's failing.. It's because my [PR on GoVector](https://github.com/drewlanenga/govector/pull/8) hasn't been merged yet.

allisonmorgan commented on README.md:
```diff
@@ -77,6 +77,14 @@ func main() {
+
+  // PushFixed method will keep the size of the Data vector constant.
+  // Oldest data points will be evicted as points are added.
+  // WARNING: Mixing Push() and PushFixed() will result in failure!
```
If `Push()` and `PushFixed()` no longer conflict, probably should remove this warning and the one below, right before the definition of `PushFixed()`.

Ropes committed:
Removing warning for problem which has been fixed and tab fix

allisonmorgan commented on anomalyze.go:
```diff
@@ -139,6 +139,18 @@ func (a *Anomalyzer) Push(x float64) float64 {
 	return a.Eval()
 }
 
+// WARNING: Mixing Push() and PushFixed() will result in failure!
```
Sorry, same thing about this comment.

Ropes replied:
woo! I should have caught that.. I think I fixed all this stuff on a separate branch but it's upstream to far to cherry pick cleanly..

allisonmorgan commented on README.md:
```diff
@@ -77,6 +77,13 @@ func main() {
+  anom2, _ := anomalyzer.NewAnomalyzer(conf, data)
+  prob2, _ := anom2.PushFixed(8.0)
+  // returns an error as second value if the array size changed unexpectantly 
+	fmt.Println("Anomalous Probability:", prob2)
```
Nit: Indentation here and I think "unexpectantly" -> "unexpectedly"

Ropes replied:
yup!

allisonmorgan commented:
Otherwise :+1:
